blivet/devices/disk.py

```diff
@@ -660,3 +660,56 @@ def dracut_setup_args(self):
                                             ":".join(opts))])
         else:
             return set(["rd.dasd=%s" % self.busid])
+
+
+class NVDIMMNamespaceDevice(DiskDevice):
+
+    """ Non-volatile memory namespace """
+    _type = "nvdimm"
+
+    def __init__(self, device, **kwargs):
+        """
+            :param name: the device name (generally a device node's basename)
+            :type name: str
+            :keyword exists: does this device exist?
+            :type exists: bool
+            :keyword size: the device's size
+            :type size: :class:`~.size.Size`
+            :keyword parents: a list of parent devices
+            :type parents: list of :class:`StorageDevice`
+            :keyword format: this device's formatting
+            :type format: :class:`~.formats.DeviceFormat` or a subclass of it
+            :keyword mode: mode of the namespace
+            :type mode: str
+            :keyword devname: name of the namespace (e.g. 'namespace0.0')
+            :type devname: str
+            :keyword sector_size: sector size of the namespace in sector mode
+            :type sector_size: str
+        """
+        self.mode = kwargs.pop("mode")
+        self.devname = kwargs.pop("devname")
+        self.sector_size = kwargs.pop("sector_size")
+
+        # UUID is not in udev so we need to use UUID we got from libblockdev data
+        _uuid = kwargs.pop("uuid")
```

Comment:
Why can't we pass the UUID from libblockdev into the `StorageDevice` ctor?

Reply:
Because `DiskDevice` constructor doesn't allow specifying UUID. But I guess I can change its constructor.

Reply:
Fixed.